src/components/Footer/Footer.tsx

```diff
@@ -0,0 +1,93 @@
+import React, { useContext } from 'react';
+import cn from 'classnames';
+import { Todo } from '../../types/Todo';
+import { Filter } from '../../types/Status';
+import { TodosContext } from '../TodosProvider';
+
+type Props = {
+  todos: Todo[];
+};
+export const Footer: React.FC<Props> = ({ todos }) => {
+  const itemsLeft = todos.filter(todo => !todo.completed).length;
+  const completed = todos.filter(todo => todo.completed);
+
+  const {
+    handleFilter,
+    filter,
+    clearAll,
+  } = useContext(TodosContext);
+
+  const handleClick = (value: Filter) => {
+    handleFilter(value);
+  };
+
+  return (
+    <footer className="todoapp__footer">
+      <span className="todo-count">
+        {`${itemsLeft} items left`}
+      </span>
+
+      <nav className="filter">
+        <a
+          href="#/"
+          className={cn(
+            'filter__link',
+            {
+              selected: filter === Filter.ALL,
+            },
+          )}
+          onClick={() => {
+            handleClick(Filter.ALL);
+          }}
+        >
+          All
+        </a>
```

Comment:
you can map enum, to prevent code duplicating

Reply:
yep, I fixed it. i thought it's more complicated but not